tests/components/discovergy/test_config_flow.py

```diff
@@ -0,0 +1,145 @@
+"""Test the Discovergy config flow."""
+from unittest.mock import patch
+
+from pydiscovergy.error import HTTPError, InvalidLogin
+
+from homeassistant import data_entry_flow, setup
+from homeassistant.components.discovergy.const import DOMAIN
+from homeassistant.config_entries import SOURCE_REAUTH, SOURCE_USER
+from homeassistant.const import CONF_EMAIL, CONF_PASSWORD
+from homeassistant.core import HomeAssistant
+
+from tests.components.discovergy import init_integration
+
+
+async def test_form(hass: HomeAssistant, mock_meters) -> None:
+    """Test we get the form."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+    assert result["type"] == data_entry_flow.FlowResultType.FORM
+    assert result["errors"] is None
+
+    with patch(
+        "homeassistant.components.discovergy.async_setup_entry",
+        return_value=True,
+    ) as mock_setup_entry:
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                CONF_EMAIL: "test@example.com",
+                CONF_PASSWORD: "test-password",
+            },
+        )
+        await hass.async_block_till_done()
+
+    assert result2["type"] == data_entry_flow.FlowResultType.CREATE_ENTRY
+    assert result2["title"] == "test@example.com"
+    assert result2["data"] == {
+        CONF_EMAIL: "test@example.com",
+        CONF_PASSWORD: "test-password",
+    }
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_reauth(hass: HomeAssistant, mock_meters) -> None:
+    """Test reauth flow."""
+    entry = await init_integration(hass)
+
+    init_result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_REAUTH, "unique_id": entry.unique_id},
+        data=None,
+    )
+
+    assert init_result["type"] == data_entry_flow.FlowResultType.FORM
+    assert init_result["step_id"] == "reauth"
+
+    configure_result = await hass.config_entries.flow.async_configure(
```

Comment:
Patch the integration setup function here too, as the config entry will be reloaded.